src/main/java/net/minecraftforge/items/CapabilityItemHandler.java

```diff
@@ -0,0 +1,283 @@
+package net.minecraftforge.items;
+
+import net.minecraft.block.Block;
+import net.minecraft.inventory.IInventory;
+import net.minecraft.inventory.ISidedInventory;
+import net.minecraft.inventory.InventoryLargeChest;
+import net.minecraft.nbt.NBTBase;
+import net.minecraft.tileentity.TileEntity;
+import net.minecraft.tileentity.TileEntityChest;
+import net.minecraft.util.BlockPos;
+import net.minecraft.util.EnumFacing;
+import net.minecraft.util.ResourceLocation;
+import net.minecraft.world.World;
+import net.minecraftforge.common.capabilities.Capability;
+import net.minecraftforge.common.capabilities.CapabilityInject;
+import net.minecraftforge.common.capabilities.CapabilityManager;
+import net.minecraftforge.common.capabilities.ICapabilityProvider;
+import net.minecraftforge.event.AttachCapabilitiesEvent;
+import net.minecraftforge.fml.common.eventhandler.EventPriority;
+import net.minecraftforge.fml.common.eventhandler.SubscribeEvent;
+import net.minecraftforge.items.wrapper.InvWrapper;
+import net.minecraftforge.items.wrapper.SidedInvWrapper;
+
+import java.util.concurrent.Callable;
+
+public class CapabilityItemHandler {
+    private static final ResourceLocation RESOURCE_LOCATION_VANILLA_CHEST = new ResourceLocation("Forge", "ItemHandler_VanillaWrapperChest");
+    private static final ResourceLocation RESOURCE_LOCATION_VANILLA_SIDED = new ResourceLocation("Forge", "ItemHandler_VanillaWrapperISided");
+    private static final ResourceLocation RESOURCE_LOCATION_VANILLA_SIMPLE = new ResourceLocation("Forge", "ItemHandler_VanillaWrapper");
+    private static final ResourceLocation RESOURCE_LOCATION_PASSTHROUGH = new ResourceLocation("Forge", "ItemHandler_PassThrough");
+    @CapabilityInject(IItemHandler.class)
+    public static Capability<IItemHandler> ITEM_HANDLER_CAPABILITY = null;
+    public static CapabilityItemHandler INSTANCE = new CapabilityItemHandler();
+
+    static
+    {
+        CapabilityManager.INSTANCE.register(IItemHandler.class, new Capability.IStorage<IItemHandler>() {
+            @Override
+            public NBTBase writeNBT(Capability<IItemHandler> capability, IItemHandler instance, EnumFacing side)
+            {
+                throw new UnsupportedOperationException();
+            }
+
+            @Override
+            public void readNBT(Capability<IItemHandler> capability, IItemHandler instance, EnumFacing side, NBTBase nbt)
+            {
+                throw new UnsupportedOperationException();
+            }
+        }, new Callable<IItemHandler>() {
+            @Override
+            public IItemHandler call() throws Exception
+            {
+                // I have no idea how to implement a default item handler
+                // maybe the empty item handler?
+                throw new UnsupportedOperationException();
```

Comment:
Yes, you would implement it with the ability to hold 0 items. Or if you feel like it 1 item, whatever you feel like it. 1 item would make the most sense considering it would show an example of how to implement it with actual functionality.